File: 20190519python/1018. Binary Prefix Divisible By 5.py
class Solution:
    def prefixesDivBy5(self, A: List[int]) -> List[bool]:
        rep=[]
        cur=0
        for num in A:
            cur=cur*2+int(num)
            if cur%5==0:
                rep.append(True)
            else:
                rep.append(False)
        return rep

File: archives/C#/1018. Binary Prefix Divisible By 5.cs
public class Solution {
    public IList<bool> PrefixesDivBy5(int[] A) {
        IList<bool> result=new List<bool>();
        int rep=0;
        for(int i=0;i<A.Length;i++){
            rep=rep*2+A[i];
            rep=rep%5;
            if(rep==0){
                result.Add(true);
            }
            else{
                result.Add(false);
            }
        }
        return result;
    }
    
}
